Source/Core/Duality/Drawing/DrawDevice.cs

```diff
@@ -519,7 +519,7 @@ public BatchInfo RentMaterial()
 			SortItem sortItem = new SortItem();
 			if (sortByDepth)
 			{
-				sortItem.SortDepth = this.CalcZSortIndex<T>(vertexBuffer, vertexCount);
+				sortItem.SortDepth = this.CalcZSortIndex<T>(slice.Data, vertexCount);
```

Comment:
Either the Z sort index needs to still be calculated based on the original vertexBuffer, or there needs to be a way to use `slice.Offset` here - otherwise the entire internal gathered vertex buffer (into which the slice refers) is used here.

Reply:
Done 👍